### PR TITLE
CB-40 Modify apiHelper to be used by components

### PR DESCRIPTION
**Pull request by femioladeji**
# JIRA TICKET NAME:
CB-40
## SUMMARY:
Modify the apiHelper.js to enable components make api requests seamlessly

### CODE REVIEW

File: app/js/helpers/apiHelper.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import fetchPolyfill from 'whatwg-fetch';
 
+const BASE_URL = '/openmrs-standalone/ws/rest/v1/';
```
Comment (dkayiwa) [inline app/js/helpers/apiHelper.js]: What happens if this is not "openmrs-standalone"?

Comment (femioladeji) [reply app/js/helpers/apiHelper.js]: Oh! I didn't think about that. I will fix that

File: app/js/helpers/apiHelper.js [hunk 2]
```diff
@@ -1,5 +1,8 @@
 import fetchPolyfill from 'whatwg-fetch';
 
+const platform = location.href.split('/')[3];
+const BASE_URL = `/${platform}/ws/rest/v1/`;
```
Comment (dkayiwa) [inline app/js/helpers/apiHelper.js]: Instead of platform, i would call it contextPath

Comment (femioladeji) [reply app/js/helpers/apiHelper.js]: Alright, I'll make the necessary adjustments now

**Comment by femioladeji** [comment]
I've made the necessary changes

File: app/js/helpers/apiHelper.js [hunk 3]
```diff
@@ -1,5 +1,8 @@
 import fetchPolyfill from 'whatwg-fetch';
 
+const contextPath = location.href.split('/')[3];
+const BASE_URL = `/${contextPath}/ws/rest/v1/`;
```
Comment (dkayiwa) [inline app/js/helpers/apiHelper.js:4]: Where will this value contextPath  come from?

Comment (femioladeji) [reply app/js/helpers/apiHelper.js:4]: The url showing on the browser will determine what the contextPath will be. For example if the user is using openmrs on http://localhost:8080/openmrs/. Then the contextPath will be `openmrs` and the `BASE_URL` for the API endpoints will be `/openmrs/ws/rest/v1/

Comment (dkayiwa) [reply app/js/helpers/apiHelper.js:4]: What i mean is that, where is the value for contextPath set from?

Comment (femioladeji) [reply app/js/helpers/apiHelper.js:4]: Ok the value of the context path is set from the address the user is visiting. The `location.href` in line 2 is a javascript inbuilt way of getting the web url that the user is on. So the contextPath is set based on the address of the cohortBuilder module that the user is accessing. I just used the `.split('/')` to specifically get if it's `openmrs` or `openmrs-standalone`

Comment (dkayiwa) [reply app/js/helpers/apiHelper.js:4]: Oh sorry i had not seen that. 😊

**Pull request merged**